tensorflow/core/kernels/cwise_op_select.cc

```diff
@@ -45,6 +45,58 @@ class SelectOp : public OpKernel {
     OP_REQUIRES_OK(ctx, ctx->input("t", &then));
     OP_REQUIRES_OK(ctx, ctx->input("e", &else_));
 
+    BCast bcast(BCast::FromShape(cond->shape()),
+                BCast::FromShape(then->shape()));
+
+    if (bcast.IsValid()) {
```

Comment:
add a comment about where this kicks in.  we currently have scalar broadcasting and primary-dimension vector broadcasting and i'd like to know if this will kick in in one of those existing cases -- because it may affect performance.